esp-bootloader-esp-idf/src/lib.rs

```diff
@@ -9,6 +9,61 @@
 //! - read the partition table
 //! - conveniently use a partition to read and write flash contents
 //!
+//! ## Examples
+//!
+//! ### Populating the Application Descriptor
+//!
+//! To use the default values:
+//!
+//! ```rust, no_run
+//! #![no_std]
+//! #![no_main]
+//!
+//! #[panic_handler]
+//! fn panic(_: &core::panic::PanicInfo) -> ! {
+//!     loop {}
+//! }
+//!
+//! esp_bootloader_esp_idf::esp_app_desc!();
+//!
+//! #[esp_hal::main]
+//! fn main() -> ! {
+//!     let _peripherals = esp_hal::init(esp_hal::Config::default());
+//!
+//!     loop {}
+//! }
+//! ```
+//!
+//! If you want to customize the application descriptor:
+//!
+//! ```rust, no_run
+//! #![no_std]
+//! #![no_main]
+//!
+//! #[panic_handler]
+//! fn panic(_: &core::panic::PanicInfo) -> ! {
+//!     loop {}
+//! }
+//!
+//! esp_bootloader_esp_idf::esp_app_desc!(
+//!     "1.0.0",
+//!     "my_project",
+//!     "12:00:00",
+//!     "2021-01-01",
+//!     "4.4",
+//!     8 * 1024,
+//!     0,
+//!     u16::MAX
```

Comment:
Can you add a comment to each line what they are? The last 4 are rather hard to understand without any help.